src/core_atmosphere/physics/mpas_atmphys_driver_convection.F

```diff
@@ -652,19 +672,29 @@ subroutine convection_from_MPAS(dt_dyn,configs,mesh,sfc_input,diag_physics,tend_
  call mpas_pool_get_array_gpu(tend_physics,'rqvcuten',rqvcuten)
  call mpas_pool_get_array_gpu(tend_physics,'rqccuten',rqccuten)
  call mpas_pool_get_array_gpu(tend_physics,'rqicuten',rqicuten)
-!$acc update host(cuprec, raincv, rthcuten, rqvcuten, rqccuten, rqicuten)
 
  do j = jts,jte
+!$acc parallel vector_length(32)
+!$acc loop gang vector
  do i = its,ite
     raincv_p(i,j) = raincv(i)
     pratec_p(i,j) = cuprec(i)
-    do k = kts,kte
-       rthcuten_p(i,k,j) = rthcuten(k,i)
-       rqvcuten_p(i,k,j) = rqvcuten(k,i)
-       rqccuten_p(i,k,j) = rqccuten(k,i)
-       rqicuten_p(i,k,j) = rqicuten(k,i)
-    enddo
  enddo
+!$acc end parallel
+ enddo
+
+ do j = jts,jte
+!$acc parallel vector_length(32)
+!$acc loop gang vector collapse(2)
+ do i = its,ite
+ do k = kts,kte
+    rthcuten_p(i,k,j) = rthcuten(k,i)
+    rqvcuten_p(i,k,j) = rqvcuten(k,i)
+    rqccuten_p(i,k,j) = rqccuten(k,i)
+    rqicuten_p(i,k,j) = rqicuten(k,i)
+ enddo
+ enddo
+!$acc end parallel
  enddo
```

Comment:
This will launch jte number of kernels. @Pranay-Reddy-Kommera @cponder Should we move the parallel region outside?